cmake/templates/python_win32_wrapper.cpp.in

```diff
@@ -87,20 +92,49 @@ int main(int argc, char* argv[]) try
         fprintf(stderr, "[DEBUG] Python script name: %s\n", scriptName.c_str());
 #endif
 
-        // use quoted string to ensure the correct path is used
-        return "\"" + scriptName + "\"";
+        return scriptName;
     };
 
-    const auto GetPythonExecutable = []() -> std::string
+    const auto GetPythonExecutable = [](const std::string& exeFullPath) -> std::string
     {
-        std::string pythonExecutable = "@PYTHON_EXECUTABLE@";
+        std::string scriptName;
+        scriptName += dirnameOf(exeFullPath);
+        scriptName += "\\@ARG_SCRIPT_NAME@";
+
+        std::ifstream infile(scriptName);
+
+        std::string firstLine;
+        if (std::getline(infile, firstLine) && firstLine.find(SHEBANG) == 0)
+        {
+            firstLine.erase(0, SHEBANG.size());
+
+            // trim extra spaces.
+            auto start = firstLine.find_first_not_of(" ");
```

Comment:
Sorry, I should have been more specific. Tabs could also be present and need to be trimmed.